src/modules/Game.class.js

```diff
@@ -23,22 +23,156 @@ class Game {
   constructor(initialState) {
     // eslint-disable-next-line no-console
     console.log(initialState);
+
+    // Ініціалізуємо гральну дошку
+    this.defaultState = [
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+    ];
+
+    this.board = initialState || JSON.parse(JSON.stringify(this.defaultState));
+    this.score = 0; // Початкови рахунок
+    this.status = 'idle'; // Початковий статус гри
+  }
+
+  moveLeft() {
+    this.checkLose();
+
+    for (let i = 0; i < 4; i++) {
+      let newRow = this.board[i].filter((value) => value !== 0);
+
+      for (let j = 0; j < newRow.length - 1; j++) {
+        if (newRow[j] === newRow[j + 1]) {
+          newRow[j] *= 2;
+          this.score += newRow[j];
+          newRow[j + 1] = 0;
+        }
+      }
+
+      newRow = newRow.filter((value) => value !== 0);
+
+      while (newRow.length < 4) {
+        newRow.push(0);
+      }
+
+      this.board[i] = newRow;
+    }
+
+    this.addRandomTile();
```

Comment:
You should not add a new tile if nothing changed:

![image](https://github.com/user-attachments/assets/7e67af06-682f-4dd2-9a34-09f09b78375e)

Then I clicked left, as result - the new tile appeared:
![image](https://github.com/user-attachments/assets/9ed38fc5-2be4-491c-90bc-5922fc34ec3f)